### PR TITLE
Add `min_weight` param to `rolling_exp` functions

### PR DESCRIPTION
**Pull request by max-sixty**

### CODE REVIEW

File: xarray/core/rolling_exp.py [hunk 1]
```diff
@@ -8,8 +8,15 @@
...
-from xarray.core.types import T_DataWithCoords, T_DuckArray
+from xarray.core.types import T_DataWithCoords
+
+try:
+    import numbagg
+    from numbagg import move_exp_nanmean, move_exp_nansum
+
+    has_numbagg = numbagg.__version__
+except ImportError:
+    has_numbagg = False
```
Comment (Illviljan) [inline xarray/core/rolling_exp.py:19]: Is this a lazy import? Does our tests catch it?

Could you add numbagg here so we don't regress with regards to #6726:

https://github.com/pydata/xarray/blob/e8be4bbb961f58ba733852c998f2863f3ff644b1/xarray/tests/test_plugins.py#L216C9-L239

You can also use this for checking module availability:

https://github.com/pydata/xarray/blob/e8be4bbb961f58ba733852c998f2863f3ff644b1/xarray/namedarray/utils.py#L81-L84

File: xarray/core/rolling_exp.py [hunk 2]
```diff
@@ -110,11 +97,16 @@ def __init__(
         obj: T_DataWithCoords,
         windows: Mapping[Any, int | float],
         window_type: str = "span",
+        min_weight: float = 0.0,
     ):
+        if has_numbagg is False or has_numbagg < "0.3.1":
+            raise ImportError("numbagg >= 0.3.1 is required for rolling_exp")
```
Comment (Illviljan) [inline xarray/core/rolling_exp.py:103]: You can use 
https://github.com/pydata/xarray/blob/e8be4bbb961f58ba733852c998f2863f3ff644b1/xarray/core/pycompat.py#L81-L83
To get the version.

Comment (max-sixty) [reply xarray/core/rolling_exp.py:103]: Though this isn't a duck array module — numba / numbagg operates on numpy arrays. Would I still use this?

File: xarray/core/rolling_exp.py (hunk 1 above)
Comment (max-sixty) [reply xarray/core/rolling_exp.py:19]: It's not a lazy import per se (it used to be, but I'm not sure that was necessarily deliberate vs. coincidental)

Using `module_available` doesn't seem to get the version?

What's the standard way of getting an optional module and checking the version in xarray? I'm happy to abide by the standard if there is one. The current approach does seem to work well in isolation...

**Comment by max-sixty** [comment]
The tests fail because the build is cached on an old version of numbagg, we can try again when that's updated

**Comment by max-sixty** [comment]
Let me know any feedback re the version handling. Otherwise ready to go, will merge in the next day or so (and then happy to make any further changes from there if we don't catch something beforehand) — hope that's reasonable

**Pull request merged**